### DISCUSSION TITLE
Auth code with Magic link

### DISCUSSION THREAD

**Discussion by fergusmeiklejohn** [supabase/supabase#3701] [Feature Requests]
Add a 2FA type code to the magic link email, so users can paste this code into the browser and login. This is useful for continuing complex UX flows where the user initiates an action (like starting a discussion, or commenting, or accessing a protected area), and that triggers authentication. With a code in the magic link, the user can copy paste this code, and continue where they left off with the action that triggered Auth.

**Comment by silentworks** [comment]
You could always add a `redirectTo` to the `.signIn` method to redirect the user back to where they were in the protected area. https://github.com/supabase/supabase/discussions/2760#discussioncomment-1144158
